### PR TITLE
Creating a new user (AUTHENTICATION)#020

### PR DESCRIPTION
**Pull request by Salim-54**
#### What does this PR do?

- Signing up a new user

#### Description of Task to be completed? 

As a user, I should be able to 
- Create a new account
- Login authorization immediately after account creation

#### How should this be manually tested?

- Clone the repo

- checkout to branch ``ft-authentication-signup-#020``

- Run `npm install` to install modules

- Run `npm  run start` to start the development server

- In your API testing tool (postman recommended) call api `localhost:3000/api/v1/users` to add a todo task

#### Any background context you want to provide?

- N/A

#### What are the relevant Pivotal Tracker/trello stories?

- [link to my pivotal tracker/tello/jira ... implemented story](https://trello.com/c/eNBFiwp2/36-signup020)

#### Screenshots (if appropriate)
- N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by Xldivin — APPROVED** [review]

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]
Remove `.env` file. it's neither committed nor pushed

File: src/controllers/blog.controller.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            new: true,
+            runValidators: true
+        })
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                blog
+            }
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:68]: Make sure you check if the blog exists before any manipulation

File: src/controllers/blog.controller.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    try{
+        
+        const blog = await Blog.findById(req.params.id);
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                blog
+            }
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:46]: Make sure you checking if the blog exists before any manipulation

File: src/controllers/blog.controller.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+exports.deleteBlogById = async (req, res) => {
+    try{
+        await Blog.findByIdAndDelete(req.params.id);
+
+        res.status(204).json({
+            status: 'success', 
+            data: null
+        });
+        // if(!blog) return res.status(404).json({success: false, message: "Blog not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/blog.controller.js:86]: Make sure you checking if the blog exists before any manipulation

File: src/controllers/query.controller.js [hunk 4]
```diff
@@ -1,15 +1,55 @@
...
-    const newQuery = new Query(query);
-    await newQuery.save();
-    res.status(201).json({success: true, data: newQuery});
-}
+exports.saveQuery  = async (req, res) => {
+
+    try{
+
+        const newQuery = await Query.create(req.body);
+        // await newQuery.save();
```
Comment (thepatrickniyo) [inline src/controllers/query.controller.js:9]: Remove the run. not good to push comments to master

File: src/controllers/query.controller.js [hunk 5]
```diff
@@ -1,15 +1,55 @@
...
+    try{
+        
+        const query = await Query.findById(req.params.id);
+        res.status(200).json({
+            status: 'success', 
+            data: {
+                query
+            }
+        });
+        // if(!query) return res.status(404).json({success: false, message: "Query not found"}) ;
```
Comment (thepatrickniyo) [inline src/controllers/query.controller.js:46]: Make sure you checking the query exists before any manipulation

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]

**Pull request merged**